### PR TITLE
Refactor read job specifications method

### PR DESCRIPTION
**Pull request by rabex028**
This pull request is conditional on the acceptance of PR #11.

We are proposing changes to the `readJobSpecifications` method in the `SpecificationReader` class because it is long and difficult to follow. To address this problem, we've split it up into two helper methods named `readTasks` and `readNumTasksForJobs` which should improve readability by showing what different segments of the code are responsible for. In the original function, there were nested loops that would read information about the specifications that the user inputs, and we separated the function at the points where the longer loops began.

Addresses #8

### CODE REVIEW

**Comment by GLOBBIMUS** [comment]
I think you have done a good job here. Splitting the method with helper methods makes easier to follow the procedure!

File: src/applications/SpecificationReader.java [hunk 1]
```diff
@@ -30,11 +30,15 @@ private void readChangeOverTimes() {
 
     private void readJobSpecifications() {
```
Comment (GLOBBIMUS) [inline src/applications/SpecificationReader.java]: Now it is easier to read this method

Comment (morrislenny) [inline src/applications/SpecificationReader.java]: I have been going back and forth on this for a while. There is no two ways about it, `readJobSpecification` in its past state, was a burning dumpster fire: It did so many different things, and it did all in one function. It was very easy to  get overwhelmed by all the logic and for loops. I really appreciate how you broke it up into several pieces, it definitely makes the athstetic of the code blocks better, but I think there is still some big areas for improvement before it gets into master. Here are my concerns:  
* *Function Names:*  Naming functions and variables is hard, and for that reason, you should think twice about creating new functions. If it can't have a short concise and meaningful name, the logic cannot be generalized and that means the code needs some heavy work before it can be its own function.  
* *Side Effects:* Almost all the functions in this class suffer from this. Moreover, this hurts the generality of the functions that you have created and makes these functions seem less dangerous than they are. For example, When I see `readTasks()`, I assume that it is going to read tasks and give them back to me, instead, it reads tasks, puts them in an array, and puts them in a class on the other side of the world, and it does it with a bunch of pointers. In reality, that functionality probably should happen in `readJobSpecification` (which is due for a new name), right where the person who is reading that function can see its side effecting nature.  

In summary, I  would like to see a few things with this: More thought about the generality of these functions, less side effects in the bowls of helper functions, and more honest variable names.

File: src/applications/SpecificationReader.java [hunk 2]
```diff
@@ -30,11 +30,15 @@ private void readChangeOverTimes() {
...
         // input the jobs
-        JobSpecification[] jobSpecifications = new JobSpecification[specification.getNumJobs()+1];
-        for (int i=1; i <= specification.getNumJobs(); i++) {
+        JobSpecification[] jobSpecifications = new JobSpecification[specification.getNumJobs() + 1];
+        for (int i = 1; i <= specification.getNumJobs(); i++) {
```
Comment (morrislenny) [inline src/applications/SpecificationReader.java]: Nice spacing! Much needed!

**Comment by rabex028** [comment]
Feedback on this pull request is reasonable, however, it is probably worth creating and attending to one or two issues before it would be ready for a merge (unless only superficial changes were made). Due to the time this would require, we will merge this PR now even though it is still in need of refactoring, as we feel an overall improvement was made.
